src/lib/modules/Document/DocumentSubjectSearch.js

```diff
@@ -0,0 +1,148 @@
+import { documentApi } from '@api/documents';
+import { Media } from '@components/Media';
+import {
+  setReactSearchKitDefaultSortingOnEmptyQueryString,
+  setReactSearchKitInitialQueryState,
+  setReactSearchKitUrlHandler,
+} from '@config';
+import history from '@history';
+import { SearchControls } from '@modules/SearchControls/SearchControls';
+import { SearchControlsOverridesMap } from '@modules/SearchControls/SearchControlsOverrides';
+import SearchFooter from '@modules/SearchControls/SearchFooter';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { OverridableContext } from 'react-overridable';
+import {
+  EmptyResults,
+  Error,
+  InvenioSearchApi,
+  ReactSearchKit,
+  ResultsLoader,
+  ResultsGrid,
+} from 'react-searchkit';
+import { Divider, Container, Ref, Sticky } from 'semantic-ui-react';
+import _isEmpty from 'lodash/isEmpty';
+import { InvenioRequestSerializer } from 'react-searchkit';
+import { DocumentSubjectGrid } from './DocumentSubjectLayout';
+import { Error as IlsError } from '@components/Error';
+import { SearchControlsMobile } from '@modules/SearchControls/SearchControlsMobile';
+
+const getSubjectsQuery = (documentMetadata) => {
+  const subjectSchemes = documentMetadata.subjects.map((subject) => {
+    return `subjects.scheme:"${subject.scheme}"`;
+  });
+  return `(${subjectSchemes.join(' OR ')})`;
+};
+
+const queryBuilderForSubjects = (documentMetadata) => {
+  let subjectsQuery = getSubjectsQuery(documentMetadata);
+
+  return class DocumentSubjectsSerielizer extends InvenioRequestSerializer {
+    serialize(stateQuery) {
+      if (_isEmpty(stateQuery.queryString)) {
+        stateQuery.queryString = subjectsQuery;
+      } else {
+        stateQuery.queryString = `${subjectsQuery} AND (${stateQuery.queryString})`;
+      }
+      stateQuery.size = 5;
+      return super.serialize(stateQuery);
+    }
+  };
+};
+
+export class DocumentSubjectSearch extends React.Component {
+  modelName = 'DOCUMENTS';
+
+  render() {
+    const { metadata } = this.props;
+    const api = new InvenioSearchApi({
+      axios: {
+        url: documentApi.searchBaseURL,
+        withCredentials: true,
+      },
+      invenio: {
+        requestSerializer: queryBuilderForSubjects(metadata),
+      },
+    });
+
+    const initialState = setReactSearchKitInitialQueryState(this.modelName);
+    const defaultSortingOnEmptyQueryString =
+      setReactSearchKitDefaultSortingOnEmptyQueryString(this.modelName);
+    const urlHandler = setReactSearchKitUrlHandler(this.modelName, false);
+    return (
+      <>
+        <Divider horizontal>More on this subject</Divider>
```

Comment:
I would try also with a regular header instead of the divider, the horizontal line does not fully fit here since there is another line above (the border of previous section)